projection/ibc_channel/view/ibc_channels.go

```diff
@@ -564,6 +565,167 @@ func (ibcChannelsView *IBCChannels) List(
 	return channels, paginationResult, nil
 }
 
+func (ibcChannelsView *IBCChannels) ListChannelsGroupByChainId(
+	order IBCChannelsListOrder,
+	filter IBCChannelsListFilter,
+	pagination *pagination.Pagination,
+) (
+	[]ChainChannels,
+	*pagination.PaginationResult,
+	error,
+) {
+
+	stmtBuilder := ibcChannelsView.rdb.StmtBuilder.Select(
+		"channel_id",
+		"port_id",
+		"connection_id",
+		"counterparty_channel_id",
+		"counterparty_port_id",
+		"counterparty_chain_id",
+		"status",
+		"packet_ordering",
+		"last_in_packet_sequence",
+		"last_out_packet_sequence",
+		"total_transfer_in_count",
+		"total_transfer_out_count",
+		"total_transfer_out_success_count",
+		"total_transfer_out_success_rate",
+		"created_at_block_time",
+		"created_at_block_height",
+		"verified",
+		"description",
+		"last_activity_block_time",
+		"last_activity_block_height",
+		"bonded_tokens",
+	).From(
+		"view_ibc_channels",
+	)
+
+	if filter.MaybeStatus != nil {
+		if *filter.MaybeStatus {
+			// Filtered channels in `opened` status
+			stmtBuilder = stmtBuilder.Where("status = ?", "true")
+		} else {
+			stmtBuilder = stmtBuilder.Where("status = ?", "false")
+		}
+	}
+
+	// MaybeLastActivityBlockTime has a higher priority than MaybeCreatedAtBlockTime
+	if order.MaybeLastActivityBlockTime != nil {
+		if *order.MaybeLastActivityBlockTime == view.ORDER_DESC {
+			stmtBuilder = stmtBuilder.OrderBy("last_activity_block_time DESC")
+		} else {
+			stmtBuilder = stmtBuilder.OrderBy("last_activity_block_time")
+		}
+	} else if order.MaybeCreatedAtBlockTime != nil {
+		if *order.MaybeCreatedAtBlockTime == view.ORDER_DESC {
+			stmtBuilder = stmtBuilder.OrderBy("created_at_block_time DESC")
+		} else {
+			stmtBuilder = stmtBuilder.OrderBy("created_at_block_time")
+		}
+	} else {
+		// By default, sort by last_activity_block_time in descending order
+		stmtBuilder = stmtBuilder.OrderBy("last_activity_block_time DESC")
+	}
+
+	rDbPagination := rdb.NewRDbPaginationBuilder(
+		pagination,
+		ibcChannelsView.rdb,
+	).BuildStmt(stmtBuilder)
+	sql, sqlArgs, err := rDbPagination.ToStmtBuilder().ToSql()
+	if err != nil {
+		return nil, nil, fmt.Errorf("error building channels select SQL: %v, %w", err, rdb.ErrBuildSQLStmt)
+	}
+
+	rowsResult, err := ibcChannelsView.rdb.Query(sql, sqlArgs...)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error executing channels select SQL: %v: %w", err, rdb.ErrQuery)
+	}
+	defer rowsResult.Close()
+
+	// key: chainID
+	// value: []IBCChannelRow
```

Comment:
I personally do not prefer to use comment for this kind of cases as we can make our code deliver the same message by:

```
type IBCChainID = string
chainChannelsMap := make(map[IBCChainID][]IBCChannelRow)
```

Reply:
The good thing about doing it with code is code always represent the latest logic; while comment relies on maintainers to pay extra care to them and update when they update the logic.

Reply:
Thanks! Agreed!